src/shared/utilities/promiseUtilities.ts

```diff
@@ -52,3 +52,8 @@ export class PromiseSharer {
 export function isThenable<T>(obj: any): obj is Promise<T> {
     return obj && typeof (<Promise<any>>obj).then === 'function'
 }
+
+/** Sleeps for the specified duration in milliseconds. Note that a duration of 0 will always wait 1 event loop. */
+export function sleep(duration: number = 0): Promise<void> {
```

Comment:
might be more discoverable in `src/shared/utilities/timeoutUtils.ts`

Reply:
Yeah wasn't sure where it could fit. From a technical point of view it makes more sense with the Promises, though it's also related to time (it's not really a timeout even though it uses `setTimeout`)